<?xml version="1.0"?>
<launch>
  <arg name="robot_ns" default="robot1"/>

  <!-- Pose Tracker -->
  <node pkg="limo_navigation" type="pose_tracker.py" name="pose_tracker" output="screen">
  </node>

  <!-- Global Planner (A*) -->
  <node pkg="limo_navigation" type="global_planner.py" name="global_planner" output="screen">
  </node>

  <!-- Local Planner (DWA) -->
  <node pkg="limo_navigation" type="local_planner.py" name="local_planner" output="screen">
    <param name="max_vel_x" value="0.5"/>
    <param name="min_vel_x" value="-0.2"/>
    <param name="max_vel_theta" value="1.0"/>
    <param name="max_acc_x" value="0.5"/>
    <param name="max_acc_theta" value="1.0"/>
    <param name="v_resolution" value="0.05"/>
    <param name="w_resolution" value="0.1"/>
    <param name="dt" value="0.2"/>
    <param name="predict_time" value="2.0"/>
    <param name="robot_radius" value="0.2"/>
    <param name="goal_cost_weight" value="1.0"/>
    <param name="speed_cost_weight" value="0.5"/>
    <param name="obstacle_cost_weight" value="2.0"/>
  </node>

  <!-- Obstacle Detector -->
  <node pkg="limo_navigation" type="obstacle_detector.py" name="obstacle_detector" output="screen">
    <param name="cluster_distance" value="0.3"/>
    <param name="min_cluster_size" value="3"/>
    <param name="max_range" value="5.0"/>
  </node>

  <!-- Frontier Explorer -->
  <node pkg="limo_navigation" type="frontier_explorer.py" name="frontier_explorer" output="screen">
    <param name="min_frontier_size" value="10"/>
    <param name="frontier_travel_point" value="0.5"/>
    <param name="exploration_rate" value="2.0"/>
  </node>

</launch>